lib/matplotlib/cbook/__init__.py

```diff
@@ -1578,6 +1578,48 @@ def simple_linear_interpolation(a, steps):
     return result
 
 
+def less_simple_linear_interpolation(x, y, xi, extrap=False):
```

Comment:
Would https://docs.scipy.org/doc/numpy/reference/generated/numpy.interp.html not be good enough?  Sorry to give you more work... :-)

Reply:
Don't worry, I'm taking this in small steps anyway since it's only going to go in 2.2 anyway. I'll have a go and see if it works.

Reply:
Not a simple drop in, so I think I'll leave it as is in this PR.